mewarpx/mewarpx/emission.py

```diff
@@ -1461,64 +1559,90 @@ class VolumeEmitter(BaseEmitter):
     """Parent class for volumetric particle injection coordinates.
 
         - ``volume`` gives the spatial volume in m^3
-        - ``_get_xv_coords()`` implements the subclass-specific particle
+        - ``_get_x_coords()`` implements the subclass-specific particle
           injection logic
     """
 
     volume = 0
+    geoms = ['Z', 'XZ', 'RZ', 'XYZ']
 
-    def __init__(self, T):
-        """Default initialization for all Emitter objects.
+    def __init__(self, T, xmin=None, xmax=None, ymin=None, ymax=None,
+                 zmin=None, zmax=None, rmin=None, rmax=None):
+        """Initialize emitter boundaries. A rectangular or cylindrical emitter
+        volume is supported. If x & y boundaries are specified the r boundaries
+        will be ignored and vice versa. If both x & y and r boundaries are
+        specified an AttributeError will be raised. If no boundaries are given
+        the simulation geometry and boundaries will be used.
 
         Arguments:
             T (float): Emitter temperature in Kelvin. Determines particle
                 velocity distribution.
+            x/y/z/rmin (float): Lower boundary of the volume.
+            x/y/z/rmax (float): Upper boundary of the volume.
         """
         super(VolumeEmitter, self).__init__()
         self.T = T
 
-    def getvoltage(self):
-        """Ideally this is probably the local potential, but default to 0."""
-        return 0.
-
-    def getvoltage_e(self):
-        """Ideally this is probably the local potential, but default to 0."""
-        return self.getvoltage()
-
+        # determine default prism type from simulation geometry
+        self.rectangular = mwxrun.geom_str != 'RZ'
```

Comment:
(1) This is now redundant with setting it in lines 1601-1604 I believe.
(2) If sticking with a bool, I would prefer a `self.cylindrical` bool rather than `self.rectangular`, because rectangular feels more like the default case. Something not being rectangular doesn't communicate to me that it's cylindrical.

Reply:
I prefer `rectangular` because rectangular feels more like the default. If we make it `self.cylindrical` then in the child classes `UniformDistributionVolumeEmitter` and `ZSinDistributionVolumeEmitter` we would have statements of the form `if not self.cylindrical` instead of `if self.rectangular`.